File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/demo/SignusDemoTest.java
package org.hyperledger.indy.sdk.demo;

import org.hyperledger.indy.sdk.IndyIntegrationTest;
import org.hyperledger.indy.sdk.pool.Pool;
import org.hyperledger.indy.sdk.pool.PoolJSONParameters;
import org.hyperledger.indy.sdk.signus.Signus;
import org.hyperledger.indy.sdk.signus.SignusResults;
import org.hyperledger.indy.sdk.signus.SignusResults.CreateAndStoreMyDidResult;
import org.hyperledger.indy.sdk.utils.PoolUtils;
import org.hyperledger.indy.sdk.wallet.Wallet;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.Timeout;

import java.util.concurrent.TimeUnit;

public class SignusDemoTest extends IndyIntegrationTest {

	@Rule
	public Timeout globalTimeout = new Timeout(1, TimeUnit.MINUTES);

	@Test
	public void testSignusDemo() throws Exception {

		//1. Create and Open Pool
		String poolName = PoolUtils.createPoolLedgerConfig();

		PoolJSONParameters.OpenPoolLedgerJSONParameter config2 = new PoolJSONParameters.OpenPoolLedgerJSONParameter(null, null, null);
		Pool pool = Pool.openPoolLedger(poolName, config2.toJson()).get();

		//2. Create and Open My Wallet
		Wallet.createWallet(poolName, "myWallet", "default", null, null).get();
		Wallet myWallet = Wallet.openWallet("myWallet", null, null).get();

		//3. Create and Open Their Wallet
		Wallet.createWallet(poolName, "theirWallet", "default", null, null).get();
		Wallet theirWallet = Wallet.openWallet("theirWallet", null, null).get();

		//4. Create My Did
		CreateAndStoreMyDidResult createMyDidResult = Signus.createAndStoreMyDid(myWallet, "{}").get();
		assertNotNull(createMyDidResult);

		//5. Create Their Did
		CreateAndStoreMyDidResult createTheirDidResult = Signus.createAndStoreMyDid(theirWallet, "{}").get();
		assertNotNull(createTheirDidResult);
		String theirDid = createTheirDidResult.getDid();
		String theirVerkey = createTheirDidResult.getVerkey();

		// 6. Store Their DID
		String identityJson = String.format("{\"did\":\"%s\", \"verkey\":\"%s\"}", theirDid, theirVerkey);
		Signus.storeTheirDid(myWallet, identityJson).get();

		// 7. Their sign message
		String msg = "{\n" +
				"        \"reqId\":1495034346617224651,\n" +
				"        \"identifier\":\"GJ1SzoWzavQYfNL9XkaJdrQejfztN4XqdsiV4ct3LXKL\",\n" +
				"        \"operation\":{\n" +
				"            \"type\":\"1\",\n" +
				"            \"dest\":\"4efZu2SXufS556yss7W5k6Po37jt4371RM4whbPKBKdB\"\n" +
				"        }\n" +
				"    }";

		byte[] signature = Signus.sign(theirWallet, theirDid, msg.getBytes()).get();

		// 8. I verify message
		Boolean valid = Signus.verifySignature(myWallet, pool, theirDid, msg.getBytes(), signature).get();
		assertTrue(valid);

		// 9. Close and delete My Wallet
		myWallet.closeWallet().get();
		Wallet.deleteWallet("myWallet", null).get();

		// 10. Close and delete Their Wallet
		theirWallet.closeWallet().get();
		Wallet.deleteWallet("theirWallet", null).get();

		//11. Close Pool
		pool.closePoolLedger().get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/Wrapper/DemoTests/SignusDemoTest.cs
using Indy.Sdk.Dotnet.Wrapper;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json.Linq;
using System.Threading.Tasks;
using static Indy.Sdk.Dotnet.Wrapper.AgentObservers;
using System;

namespace Indy.Sdk.Dotnet.Test.Wrapper.DemoTests
{
    [TestClass]
    public class SignusDemoTest : IndyIntegrationTestBase
    {
        [TestMethod]
        public void TestSignusDemo()
        {
            //1. Create and Open Pool
            var poolName = PoolUtils.CreatePoolLedgerConfig();

            var pool = Pool.OpenPoolLedgerAsync(poolName, "{}").Result;

            //2. Create and Open My Wallet
            Wallet.CreateWalletAsync(poolName, "myWallet", "default", null, null).Wait();
            var myWallet = Wallet.OpenWalletAsync("myWallet", null, null).Result;

            // 3. Create and Open Trustee Wallet
            Wallet.CreateWalletAsync(poolName, "theirWallet", "default", null, null).Wait();
            var theirWallet = Wallet.OpenWalletAsync("theirWallet", null, null).Result;

            //4. Create My Did
            var createMyDidResult = Signus.CreateAndStoreMyDidAsync(myWallet, "{}").Result;
            Assert.IsNotNull(createMyDidResult);

            //5. Create Their Did
            var createTheirDidResult = Signus.CreateAndStoreMyDidAsync(theirWallet, "{}").Result;
            Assert.IsNotNull(createTheirDidResult);
            var theirDid = createTheirDidResult.Did;
            var theirVerkey = createTheirDidResult.VerKey;

            // 6. Store Their DID
            var identityJson = string.Format("{{\"did\":\"{0}\", \"verkey\":\"{1}\"}}", theirDid, theirVerkey);
            Signus.StoreTheirDidAsync(myWallet, identityJson).Wait();

            // 7. Their sign message
            var msg = "{\n" +
                    "        \"reqId\":1495034346617224651,\n" +
                    "        \"identifier\":\"GJ1SzoWzavQYfNL9XkaJdrQejfztN4XqdsiV4ct3LXKL\",\n" +
                    "        \"operation\":{\n" +
                    "            \"type\":\"1\",\n" +
                    "            \"dest\":\"4efZu2SXufS556yss7W5k6Po37jt4371RM4whbPKBKdB\"\n" +
                    "        }\n" +
                    "    }";

            var signedMessage = Signus.SignAsync(theirWallet, theirDid, msg).Result;
            Assert.IsNotNull(signedMessage);

            // 8. I verify message
            Boolean valid = Signus.VerifySignatureAsync(myWallet, pool, theirDid, signedMessage).Result;
            Assert.IsTrue(valid);

            // 9. Close and delete My Wallet
            myWallet.CloseAsync().Wait();
            Wallet.DeleteWalletAsync("myWallet", null).Wait();

            // 10. Close and delete Their Wallet
            theirWallet.CloseAsync().Wait();
            Wallet.DeleteWalletAsync("theirWallet", null).Wait();

            //11. Close Pool
            pool.CloseAsync().Wait();
        }
       
    }
}
